backend/main.py

```
@@ -1,54 +1,348 @@
+        # --- This is the section that was missing ---
+        return QueryResponse(
+            response=response_text,
+            sources=sources,
+            document_stats=stats if request.include_sources else {},
+            model_used=rag_system.get_model_name(),
+        )
+
+    except (ValueError, TypeError) as e:
+        logger.error(f"❌ Query failed with a possible signature mismatch: {e}")
+        logger.error("💡 Hint: Ensure 'AutoRAGSystem.query' returns a tuple of (response_text, source_nodes).")
+        raise HTTPException(status_code=500, detail=f"Query processing failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)